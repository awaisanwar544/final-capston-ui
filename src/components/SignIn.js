import { Link } from 'react-router-dom';

function SignIn() {
  return (
    <div className="w-screen h-screen flex items-center bg-custom-green-500">
      <form className="bg-white shadow-xl rounded px-8 pt-6 pb-8 h-fit mx-auto max-w-xs">
        <div className="mb-4">
          <label className="block text-gray-700 text-sm font-bold mb-4" htmlFor="email">
            E-mail
            <input type="email" className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500" id="email" placeholder="Email@something.com" />
          </label>
        </div>
        <div className="mb-6">
          <label className="block text-gray-700 text-sm font-bold mb-2" htmlFor="password">
            Password
            <input className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500" id="password" type="password" placeholder="******************" />
          </label>
        </div>
        <div className="flex items-center justify-between">
          <button className="bg-custom-white-500 text-custom-grey-500 border border-custom-green-500 hover:bg-custom-green-500 hover:text-white font-bold py-2 px-4 rounded focus:outline-none" type="button">
            Sign In
          </button>
          <Link
            className="inline-block align-baseline font-bold text-sm text-custom-grey-500
            hover:text-custom-green-500"
            to="/forgot-password"
            exact
          >
            Forgot Password?
          </Link>
        </div>
      </form>
    </div>
  );
}

export default SignIn;
