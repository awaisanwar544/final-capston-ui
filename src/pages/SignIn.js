import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate, Link } from 'react-router-dom';
import logo from '../assets/logo.png';

import { logIn } from '../redux/reducers/user';

function SignIn() {
  const dispatch = useDispatch();
  const user = useSelector((state) => state.user);
  const navigate = useNavigate();

  const handleSubmit = (event) => {
    event.preventDefault();
    const email = event.target[0].value;
    const password = event.target[1].value;
    dispatch(logIn(email, password));
  };

  useEffect(() => {
    if (Object.keys(user).length !== 0) {
      localStorage.setItem('user', JSON.stringify(user));
      navigate('/');
    }
  }, [user]);
  return (
    <div className="w-screen h-screen flex items-center bg-custom-green-500">
      <form className="bg-white shadow-xl rounded px-8 pt-6 pb-8 h-fit mx-auto max-w-xs" onSubmit={handleSubmit}>
        <div className="flex w-full p-10 justify-center">
          <a href="/">
            <img className="w-24" src={logo} alt="logo" />
          </a>
        </div>
        <div className="mb-4">
          <label className="block text-gray-700 text-sm font-bold mb-4" htmlFor="email">
            E-mail
            <input type="email" className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500" id="email" placeholder="Email@something.com" />
          </label>
        </div>
        <div className="mb-6">
          <label className="block text-gray-700 text-sm font-bold mb-2" htmlFor="password">
            Password
            <input className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500" id="password" type="password" placeholder="******************" />
          </label>
        </div>
        <div className="flex items-center justify-between">
          <button className="bg-custom-white-500 text-custom-grey-500 border border-custom-green-500 hover:bg-custom-green-500 hover:text-white font-bold py-2 px-4 rounded focus:outline-none" type="submit">
            Sign In
          </button>
          <Link to="/forgot-password" exact className="inline-block align-baseline font-bold text-sm text-custom-grey-500 hover:text-custom-green-500">
            Forgot Password?
          </Link>
        </div>
      </form>
    </div>
  );
}

export default SignIn;
