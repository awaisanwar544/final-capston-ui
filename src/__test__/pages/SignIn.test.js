import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { Routes, Route } from 'react-router';
import store from '../../redux/store';
import SignIn from '../../pages/SignIn';

describe('App', () => {
  test('Render App', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" exact element={<SignIn />} />
          </Routes>
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
