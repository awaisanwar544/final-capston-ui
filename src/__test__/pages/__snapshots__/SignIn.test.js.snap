// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render App 1`] = `
<div
  className="w-screen h-screen flex items-center bg-custom-green-500"
  style={
    Object {
      "opacity": 0,
      "transform": "translateX(-100%) translateZ(0)",
    }
  }
>
  <form
    className="bg-white shadow-xl rounded px-8 pt-6 pb-8 h-fit mx-auto max-w-xs"
    onSubmit={[Function]}
  >
    <div
      className="flex w-full p-10 justify-center"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          className="w-24"
          src="logo.png"
        />
      </a>
    </div>
    <div
      className="mb-4"
    >
      <label
        className="block text-gray-700 text-sm font-bold mb-4"
        htmlFor="email"
      >
        E-mail
        <input
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500"
          id="email"
          placeholder="Email@something.com"
          required={true}
          type="email"
        />
      </label>
    </div>
    <div
      className="mb-6"
    >
      <label
        className="block text-gray-700 text-sm font-bold mb-2"
        htmlFor="password"
      >
        Password
        <input
          className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight focus:outline-custom-green-500 focus:placeholder:text-custom-green-500"
          id="password"
          placeholder="******************"
          required={true}
          type="password"
        />
      </label>
    </div>
    <div
      className="flex items-center justify-between"
    >
      <button
        className="bg-custom-white-500 text-custom-grey-500 border border-custom-green-500 hover:bg-custom-green-500 hover:text-white font-bold py-2 px-4 rounded focus:outline-none"
        type="submit"
      >
        Sign In
      </button>
      <a
        className="inline-block align-baseline font-bold text-sm text-custom-grey-500 hover:text-custom-green-500"
        exact="true"
        href="/forgot-password"
        onClick={[Function]}
      >
        Forgot Password?
      </a>
    </div>
    <a
      className="inline-block mt-10 align-baseline font-bold text-sm text-custom-grey-500 hover:text-custom-green-500"
      exact="true"
      href="/signup"
      onClick={[Function]}
    >
      Do not have an account?
      <br />
      Register now!
    </a>
  </form>
</div>
`;
